driver-core/src/main/com/mongodb/internal/operation/CreateCollectionOperation.java

```diff
@@ -361,6 +366,27 @@ private BsonDocument getCreateCollectionCommand() {
         return document;
     }
 
+    private void checkEncryptedFieldsSupported(final ConnectionDescription connectionDescription) throws MongoException {
+        if (encryptedFields != null && serverIsLessThanVersionSevenDotZero(connectionDescription)) {
+            throw new MongoClientException("Driver support of Queryable Encryption is incompatible with server."
+                    + " Upgrade server to use Queryable Encryption.");
+        }
+    }
+
+    /**
+     * @return {@code true} iff the {@linkplain #checkEncryptedFieldsSupported(ConnectionDescription) check} was successful.
+     * The {@code callback} is completed (with a failed result) iff the check was not successful.
```

Comment:
It's ["if and only if"](https://en.wikipedia.org/wiki/If_and_only_if). The statement

> The callback is completed if the check was not successful.

leaves unspecified what happens if the check was not successful. I.e., to be precise, one has to write something like

> The callback is completed if the check was not successful, otherwise the callback is not touched.

Or, one may use "iff" to write

> The callback is completed iff the check was not successful.

which is as short as the first variant, and as precise as the second one.

Reply:
Note also that the statement

> The callback is completed only if the check was not successful.

does not allow deriving whether the check was successful in a situation when the callback is not completed, while "iff" does.

Reply:
Understood, thanks for explanation!